### ISSUE TITLE
monocle.std.string.stringToLong accepts "-0" and other negative-zero forms, breaks Prism law

### ISSUE THREAD

**Issue by haskiindahouse** [julien-truffaut/Monocle#1582]
`stringToLong` violates the Prism round-trip law for `"-0"` (and a few related forms like `"-00"`, `"-01"`).

```scala
//> using scala 3.8.3
//> using dep dev.optics::monocle-core::3.3.0

import monocle.std.string.*

@main def repro(): Unit =
  val input = "-0"
  val parsed = stringToLong.getOption(input)
  println(s"getOption(\"$input\") = $parsed")

  val roundtripped = parsed.map(stringToLong.reverseGet)
  println(s"reverseGet(${parsed.get}) = ${roundtripped.get}")

  val lawHolds = roundtripped.contains(input)
  println(s"Prism law (reverseGet . getOption = id): $lawHolds")
```

Output:

```
getOption("-0") = Some(0)
reverseGet(0) = 0
Prism law (reverseGet . getOption = id): false
```

`stringToLong` should return `None` for `"-0"` because `0L.toString` is `"0"` and the round-trip law `reverseGet(getOption(s).get) == s` cannot hold.

The relevant `inputBreaksPrismLaws` checks empty string, leading `+`, and a leading `0` followed by more digits, but does not catch a leading `-0`. The same pattern produces violations on `"-00"` (parses to 0), `"-01"` (parses to -1, round-trips to `"-1"`), and so on.

Related: closed issue #244 (`String to Long / Int fails for "-"`) covers the dangling-sign case but not negative zero.

A small additional guard:

```scala
def inputBreaksPrismLaws(s: String): Boolean =
  s.isEmpty ||
  s.startsWith("+") ||
  (s.startsWith("0") && s.length > 1) ||
  (s.length > 1 && s.charAt(0) == '-' && s.charAt(1) == '0')
```

Happy to PR.

**Comment by julien-truffaut** [comment]
Thanks for the bug report @haskiindahouse. Feel free to send a PR :)
